pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -28,30 +30,45 @@ import (
 	resourceapi "k8s.io/api/resource/v1beta1"
 	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/diff"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	corev1apply "k8s.io/client-go/applyconfigurations/core/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/util/retry"
+	resourcehelper "k8s.io/component-helpers/resource"
 	"k8s.io/component-helpers/scheduling/corev1/nodeaffinity"
 	"k8s.io/dynamic-resource-allocation/cel"
 	"k8s.io/dynamic-resource-allocation/resourceclaim"
 	"k8s.io/dynamic-resource-allocation/structured"
 	"k8s.io/klog/v2"
 	fwk "k8s.io/kube-scheduler/framework"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/names"
 	schedutil "k8s.io/kubernetes/pkg/scheduler/util"
+	"k8s.io/kubernetes/pkg/util/slice"
 )
 
 const (
 	// Name is the name of the plugin used in Registry and configurations.
 	Name = names.DynamicResources
 
 	stateKey fwk.StateKey = Name
+
+	// Field manager used to update the pod status extendedResourceClaimStatus.
+	fieldManager = "KubeScheduler"
+
+	// extendedResourceClaimAnnotation is the annotation that generated special
+	// ResourceClaim get. Its value is the SHA1 hash of name of the pod for which
+	// it is generated. This is used only inside the scheduler and not documented as
+	// part of the Kubernetes API.
```

Comment:
We still need to put this into the `types.go` and document it there for API reviewers and developers. Because it is persisted in etcd, it *is* part of the Kubernetes API, even if it is used only internally.

The SHA1 is necessary because not all pod names are valid label values, right?

Can't we use a fixed `true` as label value and get the pod name from the owner references? I suspect not because we want to use the label value for efficient lookup, but it's not obvious her. [Update]: the code isn't even doing such an efficient lookup yet, although it probably could.

This is probably answered elsewhere, but let's document also here because this is what reviewers will look at first when trying to understand the purpose and semantic of this label.